snapcraft/services/package.py

```diff
@@ -122,7 +122,10 @@ def write_metadata(self, path: pathlib.Path) -> None:
             manifest = lifecycle.generate_manifest()
             manifest.to_yaml_file(snap_dir / "manifest.yaml")
 
-            shutil.copy(self._snapcraft_yaml_path, snap_dir)
+            if not self._snapcraft_yaml_path.samefile(
+                snap_dir / self._snapcraft_yaml_path.name
```

Comment:
isn't `snap_dir` inside the prime directory? Why is the snapcraft yaml there?

Reply:
It is more like a test thing where the files created in place:
```
./meta
./meta/snap.yaml
./snap
./snap/manifest.yaml
./snap/snapcraft.yaml
```

You can see it here https://github.com/canonical/snapcraft/actions/runs/8268078607/job/22620065860?pr=4662#step:6:174

Reply:
But then it looks like the test is doing something shady. @sergiusens shouldn't that test be writing into the prime dir? This is the test:

```
       def test_write_metadata_with_manifest(
          monkeypatch,
          package_service,
          default_factory,
          default_build_plan,
          new_dir,
      ):
          monkeypatch.setenv("SNAPCRAFT_BUILD_INFO", "1")
          default_factory.set_kwargs(
              "lifecycle",
              work_dir=Path("work"),
              cache_dir=new_dir,
              build_plan=default_build_plan,
          )
      
  >       package_service.write_metadata(new_dir)